### PR TITLE
Adds a step in rapidast jobs to send data to sitreps

### PR DESCRIPTION
**Pull request by san7ket**
Send report metadata to sitreps. Resolves #INSIGHTSQA-4003

cc @digitronik

### CODE REVIEW

File: vars/rapidastUtils.groovy [hunk 1]
```diff
@@ -24,6 +24,35 @@ def prepareRapidastStages(String ServiceName, String ApiScanner, String TargetUr
         stage("Collect artifacts") {
             archiveArtifacts allowEmptyArchive: true, artifacts: "results/${ServiceName}/**/zap/*.*, , results.html"
         }
+
+        stage("Send data to Sitreps Grafana") {
+            // There is a a dir that contains a timestamp which would be harder to predict, instead try to find resource.
+            def RAPIDAST_JSON_REPORT = sh returnStdout: true, script: "find . -name zap-report.json"
+            String[] str = RAPIDAST_JSON_REPORT.split("/");
+            def random_results_dir = str[3]
+            def html = "${BUILD_URL}/artifact/results/${ServiceName}/${random_results_dir}/zap/zap-report.html"
```
Comment (ntkathole) [inline vars/rapidastUtils.groovy:33]: You can use stash and unstash to get json and html report here

https://linuxhint.com/stash-in-jenkins/

File: vars/rapidastUtils.groovy [hunk 2]
```diff
@@ -24,6 +24,35 @@ def prepareRapidastStages(String ServiceName, String ApiScanner, String TargetUr
...
+            def json_path = "results/${ServiceName}/${random_results_dir}/zap/zap-report.json"
+            def raw_json = readJSON file: json_path
+            def data = [
+              service: "${ServiceName}",
+              report: raw_json,
+              html_url: html
+            ]
+            def jsonData = groovy.json.JsonOutput.toJson(data)
+            def apiUrl = "https://sitreps-backend-insights-qe-sitreps.apps.ocp-c1.prod.psi.redhat.com/api/v1/rapidast/"
```
Comment (ntkathole) [inline vars/rapidastUtils.groovy:42]: Let's add this in https://github.com/RedHatInsights/insights-pipeline-lib/blob/master/vars/pipelineVars.groovy

File: vars/rapidastUtils.groovy [hunk 3]
```diff
@@ -24,6 +24,35 @@ def prepareRapidastStages(String ServiceName, String ApiScanner, String TargetUr
...
+            ]
+            def response = httpRequest(
+                url: apiUrl,
+                httpMode: 'PUT',
+                requestBody: jsonData,
+                headers: headers
+            )
+            echo "Response status: ${response.status}"
+            echo "Response body: ${response.content}"
+        }
```
Comment (ntkathole) [inline vars/rapidastUtils.groovy:55]: do we want job to fail if it failed to send report to sitreps ?

Comment (digitronik) [reply vars/rapidastUtils.groovy:55]: I don't think we need to fail job if sending fails.

**Review by digitronik — APPROVED** [review]
LGTM :+1: 
Please address @ntkathole comment.

File: vars/rapidastUtils.groovy (hunk 1 above)
Comment (san7ket) [reply vars/rapidastUtils.groovy]: we are not trying to send the report, but the url of the HTML report. So, we would need to know the path for viewable html url

Comment (ntkathole) [reply vars/rapidastUtils.groovy]: Ok, I think more simpler way here is to use https://plugins.jenkins.io/htmlpublisher/, which will give you the direct url of html file.

use `publishHTML()` in colelct artifacts step and you will get your url like `${BUILD_URL}/zap-report.html`, which will remove the workaround that you used to get url.

**Review by ntkathole — APPROVED** [review]
ACK. Looks good 👍

**Review by bsquizz — APPROVED** [review]

**Pull request merged**